traits/trait_list_object.py

```diff
@@ -53,460 +88,750 @@ def __init__(self, index=0, removed=None, added=None):
+            added : list
+                The  new values that were added.
+            removed : list
+                The empty list [].
+
+        """
+        self.extend(other)
+        return self
 
-    def pop(self, *args):
-        if not hasattr(self, "trait"):
-            return list.pop(self, *args)
-        if self.trait.minlen < len(self):
-            if len(args) > 0:
-                index = args[0]
+    def __imul__(self, count):
+        """ Implements the self *= value operator.
+
+        Parameters
+        ----------
+        count : int
+            Number of times to duplicate the list.
+
+        Notes
+        -----
+        Notification:
+            index : slice
+                Slice ranging from current_length to new_length.
+            added : list
+                The  new values that were added.
+            removed : list
+                The empty list [].
+
+        """
+        if count > 1:
+            self.extend(self * (count - 1))
+        elif count == 0:
+            self[:] = []
+        return self
+
+    # ------------------------------------------------------------------------
+    # Private interface
+    # ------------------------------------------------------------------------
+
+    def _get_removed(self, index):
+        """ Compute removed values given index. """
+        try:
+            return self[index]
+        except Exception:
+            if isinstance(index, slice):
+                return []
             else:
-                index = -1
+                return Undefined
 
-            try:
-                removed = [self[index]]
-            except Exception:
-                pass
+    def _normalize_index(self, index):
+        """ Normalize integer index to range 0 to len (inclusive). """
+        index = operator.index(index)
+        if index < 0:
+            return max(0, len(self) + index)
+        else:
+            return min(len(self), index)
 
-            result = list.pop(self, *args)
+    def _normalize_slice(self, index):
+        """ Normalize slice start and stop to range 0 to len (inclusive). """
+        if index.step is None or index.step > 0:
+            if index.start is not None:
+                start = self._normalize_index(index.start)
+            else:
+                start = 0
+            if index.stop is not None:
+                stop = self._normalize_index(index.stop)
+            else:
+                stop = len(self)
+        else:
+            if index.start is not None:
+                start = self._normalize_index(index.start)
+            else:
+                start = len(self)
+            if index.stop is not None:
+                stop = self._normalize_index(index.stop)
+            else:
+                stop = 0
 
-            if self.name_items is not None:
-                if index < 0:
-                    index = len(self) + index + 1
+        return slice(start, stop, index.step)
```

Comment:
Yes it can! I didn't know about the `indices` method. Thanks!

Reply:
I am sorry for the ill advice. The returned value of `slice.indices` cannot used directly as input arguments to `slice` when the `step` is negative, so the negative step case needs to be handled separately. The original implementation of `_normalize_slice` still seems to me a bit too complicated though.

Reply:
I am starting to think this is not worth the effort.  It may be best to leave the slice as it is (which was the existing behaviour anyway), instead of risk sending a wrong value.